wireless_modules/sensors/gps_sensor.py

```diff
@@ -0,0 +1,60 @@
+from machine import UART
+import uasyncio as asyncio
+
+from micropyGPS import MicropyGPS
+
+from sensor_base import Sensor
+
+
+class GpsSensor(Sensor):
+    """
+    A GPS sensor implementing NMEA-0183 connected via UART, such as the u-blox NEO 6M.
+    """
+
+    def __init__(self, uart_channel: int):
+        """
+        Begin reading data from the GPS over UART.
+        :param uart_channel: The UART channel that the GPS is connected to.
+        """
+        self.gps = MicropyGPS(location_formatting="dd")
+        asyncio.create_task(self.uart_rx(uart_channel))
+
+    async def uart_rx(self, uart_channel: int):
+        """
+        Read and load data from the GPS over UART.
+        """
+        uart = UART(uart_channel, 9600)
+        stream_reader = asyncio.StreamReader(uart)
+        while True:
+            res = await stream_reader.readline()
+            for char in res:
+                self.gps.update(chr(char))
+
+    def read(self):
+        """
+        Get the most recently stored GPS data.
+        :return: An array containing the GPS sensor dictionary. `value` in this
+        dictionary contains spacial, quality of service and time information.
+        """
+        latitude = self.gps.latitude[0] * (1 if self.gps.latitude[1] == "N" else -1)
+        longitude = self.gps.longitude[0] * (1 if self.gps.longitude[1] == "E" else -1)
+        # ISO 8601 datetime format
+        datetime = "20{:02d}-{:02d}-{:02d}T{:02d}:{:02d}:{:02.0f}".format(
+            self.gps.date[2], self.gps.date[1], self.gps.date[0], *self.gps.timestamp
```

Comment:
Not necessarily for a particular reason, but I thought it had potential to be useful in the future. The biggest use I see it potentially having is so we can work out when a log file is from, as the files are not saved with any meaningful timestamp.

Reply:
But then do you think we need to publish it every second? We already have the traveled distance data so we can  graph anything over time.

Reply:
But I guess it doesn't cost us much to publish it every second, so if we think it's alright, I guess it's fine!

Reply:
@hallgchris collecting date and time from the GPS would be very handy for logging. I think it would solve the problem with not having a real time clock.